core/src/test/scala/unit/kafka/server/OffsetCommitRequestTest.scala

```diff
@@ -0,0 +1,163 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.server
+
+import kafka.test.ClusterInstance
+import kafka.test.annotation.{ClusterConfigProperty, ClusterTest, ClusterTestDefaults, Type}
+import kafka.test.junit.ClusterTestExtensions
+import kafka.test.junit.RaftClusterInvocationContext.RaftClusterInstance
+import kafka.test.junit.ZkClusterInvocationContext.ZkClusterInstance
+import kafka.utils.TestUtils
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.{Tag, Timeout}
+import org.junit.jupiter.api.extension.ExtendWith
+
+import java.util.stream.Collectors
+import scala.jdk.CollectionConverters._
+
+@Timeout(120)
+@ExtendWith(value = Array(classOf[ClusterTestExtensions]))
+@ClusterTestDefaults(clusterType = Type.KRAFT, brokers = 1)
+@Tag("integration")
+class OffsetCommitRequestTest(cluster: ClusterInstance) extends GroupCoordinatorBaseRequestTest(cluster) {
+
+  @ClusterTest(serverProperties = Array(
+    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "true"),
+    new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+    new ClusterConfigProperty(key = "group.consumer.max.session.timeout.ms", value = "600000"),
```

Comment:
Is this the current way we enable setting new consumer group protocol? Setting these two configs?

Reply:
Yeah, at the moment, you have to use `unstable.api.versions.enable` and `group.coordinator.new.enable`. This will change when we release it though. It will be based on the metadata version.

Reply:
Sorry I didn't ask this clearly.  For `testOffsetCommitWithNewConsumerGroupProtocolAndNewGroupCoordinator` and `testOffsetCommitWithOldConsumerGroupProtocolAndNewGroupCoordinator` the only different was the timeout configs:

```
new ClusterConfigProperty(key = "group.consumer.max.session.timeout.ms", value = "600000")
new ClusterConfigProperty(key = "group.consumer.session.timeout.ms", value = "600000")
```

Is this how we enable the new group protocol? Both methods set `unstable.api.versions.enable` and `group.coordinator.new.enable` to be true. Should the second one have the api versions be false?

Reply:
The two settings that you mentioned are regular configs for the new protocol. They don't enable the new protocol. As I said in my previous comment, the new protocol is automatically enabled when `unstable.api.versions.enable` and `group.coordinator.new.enable` are set at the moment. Later, the metadata version will be used.

Reply:
I feel like maybe I'm missing something obvious but the test is called `testOffsetCommitWithOldConsumerGroupProtocolAndNewGroupCoordinator()` so it says we use the old protocol.

But 
> the new protocol is automatically enabled when unstable.api.versions.enable and group.coordinator.new.enable are set at the moment.

So should one of them be false so we have the old consumer group protocol?

Reply:
```
  def testOffsetCommitWithOldConsumerGroupProtocolAndNewGroupCoordinator(): Unit = {
    testOffsetCommit(false)
  }
```

Whether the new or the old protocol is used is defined by the parameter passed to `testOffsetCommit`. It is not used automatically because it is available. I need to set `unstable.api.versions.enable` here to test the unreleased versions of the offset commit/fetch apis. I know.. This is confusing...

Reply:
Ok -- got it. Thanks for clarifying. That's what I was looking for.